src/CCTPBase.sol

```diff
@@ -0,0 +1,169 @@
+        USDC = _USDC;
+    }
+
+    function getCCTPDomain(uint16 chain) internal pure returns (uint32) {
+        if (chain == 2) {
+            return 0;
+        } else if (chain == 6) {
+            return 1;
+        } else if (chain == 23) {
+            return 3;
+        } else if (chain == 24) {
+            return 2;
+        } else {
+            revert("Wrong CCTP Domain");
+        }
+    }
+
+    function redeemUSDC(bytes memory cctpMessage) internal returns (uint256 amount) {
+        (bytes memory message, bytes memory signature) = abi.decode(cctpMessage, (bytes, bytes));
+        uint256 beforeBalance = IERC20(USDC).balanceOf(address(this));
+        circleMessageTransmitter.receiveMessage(message, signature);
+        return IERC20(USDC).balanceOf(address(this)) - beforeBalance;
+    }
+}
+
+abstract contract CCTPSender is CCTPBase {
+    uint8 internal constant CONSISTENCY_LEVEL_FINALIZED = 15;
+
+    using CCTPMessageLib for *;
+
+    /**
+     * transferTokens wraps common boilerplate for sending tokens to another chain using IWormholeRelayer
+     * - approves tokenBridge to spend 'amount' of 'token'
+     * - emits token transfer VAA
+     * - returns VAA key for inclusion in WormholeRelayer `additionalVaas` argument
+     *
+     * Note: this requires that only the targetAddress can redeem transfers.
+     *
+     */
+
+    function transferUSDC(uint256 amount, uint16 targetChain, address targetAddress)
+        internal
+        returns (MessageKey memory)
+    {
+        IERC20(USDC).approve(address(circleTokenMessenger), amount);
+        uint64 nonce = circleTokenMessenger.depositForBurnWithCaller(
+            amount,
+            getCCTPDomain(targetChain),
+            addressToBytes32CCTP(targetAddress),
+            USDC,
+            addressToBytes32CCTP(targetAddress)
+        );
+        return MessageKey(
+            CCTPMessageLib.CCTP_KEY_TYPE, abi.encodePacked(getCCTPDomain(wormhole.chainId()), nonce)
+        );
+    }
+
+    function sendUSDCWithPayloadToEvm(
+        uint16 targetChain,
+        address targetAddress,
+        bytes memory payload,
+        uint256 receiverValue,
+        uint256 gasLimit,
+        uint256 amount
+    ) internal returns (uint64 sequence) {
+        MessageKey[] memory messageKeys = new MessageKey[](1);
+        messageKeys[0] = transferUSDC(amount, targetChain, targetAddress);
+
+        bytes memory userPayload = abi.encode(amount, payload);
+        address defaultDeliveryProvider = wormholeRelayer.getDefaultDeliveryProvider();
+
+        (uint256 cost,) = wormholeRelayer.quoteEVMDeliveryPrice(targetChain, receiverValue, gasLimit);
+        
+        sequence = wormholeRelayer.sendToEvm{value: cost}(
+            targetChain,
+            targetAddress,
+            userPayload,
+            receiverValue,
+            0,
+            gasLimit,
+            targetChain,
+            address(0x0),
```

Comment:
I think for simplicity - if the default behavior is to send to any address specifically on the target chain, then it might show a 'execution reverted' message on the target chain blockscan website if that address doesn't accept payments (e.g. if it is a contract that didn't implement a fallback function for receiving value)